File: BackEnd/FoodOrder.Core/ViewModels/Foods/FoodVM.cs
﻿using FoodOrder.Core.Models;
using FoodOrder.Core.ViewModels.Categories;
using FoodOrder.Core.ViewModels.SaleCampaigns;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FoodOrder.Core.ViewModels.Foods
{
    public class FoodVM
    {
        public int ID { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public double Price { get; set; }
        public int Count { get; set; }
        public string ImagePath { get; set; }

        public double AgvRating { get; set; }
        public int TotalRating { get; set; }

        public bool IsDeleted { get; set; }
        public DateTime? TimeDeleted { get; set; }


        public SaleCampaignVM SaleCampaignVM { get; set; }

        public List<CategoryVM> CategoryVMs { get; set; }

        //public List<OrderDetail> OrderDetails { get; set; }
        public List<Rating> Ratings { get; set; }
        public List<Cart> Carts { get; set; }
    }
}

File: bot/bot/actions/vm/FoodVM.py
class FoodVM:
    def __init__(self, 
        id,
        name,
        description,
        price,
        count,
        imagePath,
        agvRating,
        totalRating,
        saleCampaignVM,
        categoryVMs,
        ratings,
       carts):
       self.id=id
       self.name=name
       self.description=description
       self.price=price
       self.count=count
       self.imagePath=imagePath
       self.agvRating=agvRating
       self.totalRating=totalRating
       self.saleCampaignVM=saleCampaignVM
       self.categoryVMs=categoryVMs
       self.ratings=ratings
       self.carts=carts